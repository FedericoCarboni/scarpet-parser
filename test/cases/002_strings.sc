// String test
'plain string';
'escape sequences \n \t \' \\ plain text';
'string
spanning
multiple
lines'
