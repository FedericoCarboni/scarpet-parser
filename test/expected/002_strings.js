/** @type {import('../../lib/Parser.js').Node} */
export const root = {
    kind: 'BinaryExpression',
    operator: ';',
    lvalue: {
        kind: 'BinaryExpression',
        operator: ';',
        lvalue: {
            kind: 'StringLiteral',
            literal: "'plain string'",
            value: 'plain string',
            start: {
                offset: 15,
                line: 1,
                character: 0,
            },
            end: {
                offset: 29,
                line: 1,
                character: 14,
            },
        },
        rvalue: {
            kind: 'StringLiteral',
            literal: "'escape sequences \\n \\t \\' \\\\ plain text'",
            value: "escape sequences \n \t ' \\ plain text",
            start: {
                offset: 31,
                line: 2,
                character: 0,
            },
            end: {
                offset: 72,
                line: 2,
                character: 41,
            },
        },
        comment: '// String test\n',
        start: {
            offset: 15,
            line: 1,
            character: 0,
        },
        end: {
            offset: 72,
            line: 2,
            character: 41,
        },
    },
    rvalue: {
        kind: 'StringLiteral',
        literal: "'string\nspanning\nmultiple\nlines'",
        value: 'string\nspanning\nmultiple\nlines',
        start: {
            offset: 74,
            line: 3,
            character: 0,
        },
        end: {
            offset: 106,
            line: 6,
            character: 6,
        },
    },
    comment: null,
    start: {
        offset: 15,
        line: 1,
        character: 0,
    },
    end: {
        offset: 106,
        line: 6,
        character: 6,
    },
};
