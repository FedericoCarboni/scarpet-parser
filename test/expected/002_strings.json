{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "operator": ";",
    "lvalue": {
      "kind": "BinaryExpression",
      "operator": ";",
      "lvalue": {
        "kind": "StringLiteral",
        "literal": "'plain string'",
        "value": "plain string",
        "start": {
          "pos": 15,
          "row": 1,
          "col": 0
        },
        "end": {
          "pos": 29,
          "row": 1,
          "col": 14
        }
      },
      "rvalue": {
        "kind": "StringLiteral",
        "literal": "'escape sequences \\n \\t \\' \\\\ plain text'",
        "value": "escape sequences \n \t ' \\ plain text",
        "start": {
          "pos": 31,
          "row": 2,
          "col": 0
        },
        "end": {
          "pos": 72,
          "row": 2,
          "col": 41
        }
      },
      "comment": "// String test\n",
      "start": {
        "pos": 15,
        "row": 1,
        "col": 0
      },
      "end": {
        "pos": 72,
        "row": 2,
        "col": 41
      }
    },
    "rvalue": {
      "kind": "StringLiteral",
      "literal": "'string\nspanning\nmultiple\nlines'",
      "value": "string\nspanning\nmultiple\nlines",
      "start": {
        "pos": 74,
        "row": 3,
        "col": 0
      },
      "end": {
        "pos": 106,
        "row": 6,
        "col": 6
      }
    },
    "comment": null,
    "start": {
      "pos": 15,
      "row": 1,
      "col": 0
    },
    "end": {
      "pos": 106,
      "row": 6,
      "col": 6
    }
  }
}